deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -303,6 +343,12 @@ boolean alreadyExpired() {
         }
     }
 
+    private enum EnforcementState {
+        NOT_ENFORCED,
+        ENFORCED_THROW,
+        ENFORCED_DONT_THROW
```

Comment:
what does ENFORCED_DONT_THROW mean?

Reply:
that we need to pass propagate a header to enforce the deadline, but we shouldn't throw if the deadline has expired.

Reply:
hmm, is this coming from the thread here? https://github.com/palantir/deadlines-java/pull/34#discussion_r1964053775